### ISSUE TITLE
Move dev tests from `npm test` to `gulp test:dev`.

### ISSUE THREAD

**Issue by szymonkups** [ckeditor/ckeditor5#167]
Tests for our development tasks are executed by
```
npm test
```
Let's unify them with testing under Node.js, and move them to gulp:
```
gulp test:dev
gulp test:dev:coverage
```

They can be added to gulp task under `dev/tasks/test/tasks.js` when #84 issue will be closed.

**Comment by pomek** [comment]
All the tests pass when they are run from npm scripts. If I wanted to use Gulp for this, package [gulp-mocha](https://www.npmjs.com/package/gulp-mocha) won't collaborate with me. I guess, the reason is neested Gulp. Inside  Gulp I run another Gulp and maybe this is a problem. 

But, if Mocha will be running in [separate process](https://www.npmjs.com/package/gulp-spawn-mocha), everything works properly. 

I will change `gulp-mocha` instead of `gulp-spawn-mocha`.

**Comment by pomek** [comment]
TL;DR:
* `gulp test:dev` fails (_lint lintStaged should throw error when linting fails_)
* `gulp test:dev:coverage` generate coverage report, but this tasks falls into infinity loop

---

It's more complicated than I would suppose. I abandoned package `gulp-spawn-mocha`, because it does not help.

Generally, a problem occurs when some Gulp module runs another Gulp module. I don't understand what's actually happening, so I cannot describe (or explain) the problem.

Mocha tests pass when they are running via npm scripts. If it's running via Gulp, they fail. But, @szymonkups noticed one thing. If sequence of Gulp is clear ([`gulp.seq = [];`](https://github.com/ckeditor/ckeditor5/compare/master...t/167#diff-5a4fa2ea639e92de3f3215e4e853480dR127)), coverage report is generate properly. Unfortunately, it looks like hack, not like a solution.

Without the line, Gulp task `test:dev:coverage` falls into infinity loop. I thought about something like this:

```js
devTest(done) {
	return gulp.src( 'dev/tests/**/*.js' )
		.pipe( mocha() )
		.pipe( tasks.coverage ? istanbul.writeReports() : buildUtils.noop() )
		.on( 'end', done );
}
```

but, it throws error:

```
[15:39:58] Error: task completion callback called too many times
    at finish (/Users/pomek/Projects/cksource/ckeditor5/node_modules/orchestrator/lib/runTask.js:15:10)
    at /Users/pomek/Projects/cksource/ckeditor5/node_modules/orchestrator/lib/runTask.js:52:4
    at f (/Users/pomek/Projects/cksource/ckeditor5/node_modules/once/once.js:17:25)
    at DestroyableTransform.onend (/Users/pomek/Projects/cksource/ckeditor5/node_modules/end-of-stream/index.js:31:18)
    at emitNone (events.js:91:20)
    at DestroyableTransform.emit (events.js:185:7)
    at endReadableNT (/Users/pomek/Projects/cksource/ckeditor5/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:865:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

@szymonkups suggested, transfer all logic from Gulp modules into another files (which can be tested). Gulp modules will require these files. We don't want to have tests for integration between these modules and Gulp modules. Only separated logic will have tests. 
I don't know whether I explained it good enough. If I'm wrong, @szymonkups, correct me.

**Comment by Reinmar** [comment]
This is incorrect:

```js
devTest(done) {
    return gulp.src( 'dev/tests/**/*.js' )
        .pipe( mocha() )
        .pipe( tasks.coverage ? istanbul.writeReports() : buildUtils.noop() )
        .on( 'end', done );
}
```

You cannot listen to the stream#end event and at the same time returning the stream from the function. Do just one thing (return the stream).

**Comment by szymonkups** [comment]
I did some short research on this. It looks like some kind execution loop happens here. We start test tasks from `dev/tasks/test/tasks.js` and then this file is also included in tests (we are testing if test methods works fine) and somehow task is executed once more.
